apollo-gradle-plugin-incubating/src/main/kotlin/com/apollographql/apollo/gradle/api/CompilationUnit.kt

```diff
@@ -17,6 +17,12 @@ interface CompilationUnit {
   val outputDir: Provider<Directory>
   val transformedQueriesDir: Provider<Directory>
 
+  fun configure(configure: Action<Params>)
   fun setSources(rootFolder: Provider<Directory>)
-  fun setSources(rootFolders: FileCollection, graphqlFiles: FileCollection, schemaFile: Provider<RegularFile>, rootPackageName: Provider<String>)
+
+  data class Params(
+      val graphqlFolder: DirectoryProperty,
```

Comment:
Maybe make this a `SourceDirectorySet` so we can add multiple folders and add ìncludes`/`excludes` ?

Reply:
The problem with that is that `SourceDirectorySet` and `AndroidSourceDirectorySet` are 2 separate classes with 100% same API surface. It is hard to support both.

If we would have full `SourceSet` support, then this would make sense. But now, it is also hard to create a new `SourceDirectorySet` since it is created by the official plugins.